service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java

```diff
@@ -36,12 +37,18 @@
     @Qualifier("sampleSparkRepository")
     private SampleRepository sampleSparkRepository;
     @Autowired
+    private CopyNumberSegmentSparkRepository copyNumberSegmentSparkRepository;
```

Comment:
Is this not implementing the CopyNumberSegmentRepos?

Reply:
This is another case where the method returns internalIds and filters based on that. (`List<Integer> samplesWithCopyNumberSeg = copyNumberSegmentRepository.fetchSamplesWithCopyNumberSegments()`) So the workaround was to create a method that returns `List<String> sampleIds` and create another method in SampleSeriviceImpl.

Reply:
Why not introduce a method to the interface which returns a List<String> (or return List<Sample> and have the service process internal ids or stable sample ids as required)?

Reply:
thanks for the suggestion. I added a method in the interface that returns a list of Samples.